### PR TITLE
Add suggested gems and config for Heroku

### LINKED ISSUE #11: Heroku Configuration
Heroku recommends adding the `rails_12factor` gem in production and using a `Procfile` to specify a web server that is not webrick.

### PR DESCRIPTION
**Pull request by dlackty**
Close #11

### CODE REVIEW

File: config/puma.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+workers Integer(ENV['WEB_CONCURRENCY'] || 2)
+threads_count = Integer(ENV['MAX_THREADS'] || 5)
```
Comment (KrauseFx) [inline config/puma.rb]: Are these the correct defaults? I don't think it makes sense to run `boarding` on more than one dyno.

Comment (dlackty) [inline config/puma.rb]: Yes. I actually copied this config from [Heroku's documentation](https://devcenter.heroku.com/articles/deploying-rails-applications-with-the-puma-web-server), and this config is for Puma's concurrency setting and has nothing to do with dyno.

**Comment by KrauseFx** [comment]
Fair enough, thanks @dlackty :+1:

**Comment by KrauseFx** [comment]
Reverted with https://github.com/fastlane/boarding/commit/30c97cd9152d24aa54bb8b6547bc61c496b72978 because of https://github.com/fastlane/boarding/issues/26

**Pull request merged**